packages/react-tinacms-editor/src/types.ts

```diff
@@ -0,0 +1,22 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+export interface ImageProps {
+  upload?: () => Promise<string>[]
```

Comment:
I think this will end up being:

```
  upload?: () => Promise<string[]>
```

Reply:
Can you plz confirm, I can refactor code accordingly.

Reply:
[Yup](https://github.com/tinacms/tinacms/blob/master/packages/@tinacms/media/src/media.ts#L59).

MediaStore returns array of Media objects. These will be passed as strings:

```js
upload: async (files) => {
  const media = await cms.media.store.persist(files)
  return media.map(toSrc)
}
```

Reply:
Hey @ncphillips , PR is updated.